sinks/api/types.go

```diff
@@ -148,24 +153,35 @@ type SupportedStatMetric struct {
 	GetValue func(*cadvisor.ContainerSpec, *cadvisor.ContainerStats) []internalPoint
 }
 
-// Timeseries represents a single metric.
+// Timeseries represents a set of points for a series.
 type Timeseries struct {
-	Point            *Point
-	MetricDescriptor *MetricDescriptor
+	// The name of the series.
+	SeriesName string
+
+	// Precision of points timestamps.
+	TimePrecision TimePrecision
+
+	// The labels used by this series.
+	LabelKeys []string
```

Comment:
Yes, it is an optimization. Each `timeseries` can have multiple points, and any given point may specify some subset of labels for the timeseries. So, instead of iterating through all points and their lables to figure out the complete set of label keys, we have `LabelKeys` defined at the top defining all the labels that points in this `timeseries` may have (if any individual point defines a label that is not part of `labelKeys`, it is ignored, if any individual point does not provide a value for a label that is part of `labelKeys`, it is left empty).